api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -362,9 +362,14 @@ public DomainValuePair apply(DataSetValues dataSetValue) {
                 "Timeout while querying the CDR to pull preview information.");
           }
 
+          final List<String> requestValues =
+              dataSet.getValues().stream()
+                  .map(DomainValuePair::getValue)
+                  .collect(Collectors.toList());
+
           Collections.sort(
               valuePreviewList,
-              Comparator.comparing(item -> dataSet.getValues().indexOf(item.getValue())));
+              Comparator.comparing(item -> requestValues.indexOf(item.getValue())));
```

Comment:
No `DomainValuePair` matched the `String` so `indexOf()` was always -1 and therefore no sorting happened here.

Reply:
so this is something I was wondering aloud the other day: should the queries have a canonical form so they're mostly repeatable, or do we hold the position that you could get a different, equivalent query the next time you use the builder. Might be amusing for workshops if everybody's query was sorted differently.

Reply:
Let's chat about refactoring the nomenclature here. I hate using value as a name for anything, especially a column. If nothing else, we can keep it in the UI but make the code base easier to unersdand.